pkg/summary/summarizers.go

```diff
@@ -204,6 +206,22 @@ func initializeCheckErrors() {
 	logrus.Debugf("GVK Error Mapping not provided, using predefined values")
 }
 
+func checkGeneration(obj data.Object, _ []Condition, summary Summary) Summary {
+	if summary.Attributes["hasObservedGeneration"] == true {
```

Comment:
Do we need this attribute? Can't we look at .status.observedGeneration from the obj directly to know if it has one? eg: if it's set, then it has one, if not, it doesn't?

Reply:
>No, when the object is just created, it won't have any status even though the CRD has status and ObservedGeneration field defined. 

Right, so can't we detect that case from the object directly, set status to in-progress (or unknown, I'm not sure what the states we have are).

Then, when the status is added, our controllers should receive that information and we'll just recalculate the summary ?

Reply:
Sorry can you elaborate on this ? 

How can I detect that case from the object directly ? the obj is something like an instance of the CRD. It has no informtion about the definition.

Reply:
Something like:

```go
	_, hasObservedGeneration := data.GetValue(obj, "status", "observedGeneration")

```

`data` module is from wrangler (pkg/data). (obj is a `map[string]any`)

Reply:
It is still returning `false` 

Here is what I tested. 

<img width="685" height="193" alt="Screenshot 2025-07-21 at 1 45 30 PM" src="https://github.com/user-attachments/assets/a3dd81e5-1e81-4d3b-a94b-aedf4a0b6951" />


<img width="430" height="647" alt="Screenshot 2025-07-21 at 1 45 49 PM" src="https://github.com/user-attachments/assets/e9c71f4c-5b56-492d-bb82-17e8c9d65fb9" />

In fact, ClusterRepo CRD has `ObservedGeneration`. 

Note: Once the controller has processed this, the above code will return true becuase, the controller will put the status after it processes it.